### PR TITLE
Update styling for the side nav

### LINKED ISSUE #8685: Add tooltip for Expand,Profile and Logout
**Is your feature request related to a problem? Please describe.**
Currently there is no tooltip for the " expand,Profile and Logout" icons when we use minimized sidebar option. this makes the users accidentally click logout instead of expand icon. 
**Describe the solution you'd like**
Add tooltip for these three icons 

- [ ] Expand
- [ ] Profile
- [ ] Logout

**Describe alternatives you've considered**

WhatsApp Image 2024-10-01 at 15 56 33_c8cd3648

### PR DESCRIPTION
**Pull request by vinutv**
@ohcnetwork/care-fe-code-reviewers

- Closes #8685

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Common/Sidebar/SidebarItem.tsx
Comment (sainak) [inline src/Components/Common/Sidebar/SidebarItem.tsx:1]: line 63: if we add `text-nowrap` it will remove the text wrapping during the transition

**Review by bodhish — APPROVED** [review]
@nihal467 do fix the specs here

Also lets gets this merged asap

File: src/Components/Common/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -215,17 +224,26 @@ interface ToggleShrinkProps {
...
+          <CareIcon
+            icon={shrinked ? "l-arrow-bar-right" : "l-layout-sidebar-alt"}
+            className={`text-lg ${
+              shrinked ? "" : ""
+            } transition-all delay-150 duration-300 ease-out`}
+          />
+        </button>
+      </TooltipTrigger>
+      <TooltipContent>
+        <p>{shrinked ? "Expand Sidebar" : "Collapse Sidebar"}</p>
```
Comment (vinutv) [inline src/Components/Common/Sidebar/Sidebar.tsx]: @nihal467 Could you help update this with i18n translation?

File: src/Components/Common/Sidebar/SidebarItem.tsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ const SidebarItemBase = forwardRef(
           <span
             className={`${
               shrinked ? "hidden" : "grow"
-            } flex w-full items-center pl-4 text-sm tracking-wide`}
+            } flex w-full items-center text-nowrap pl-4 text-sm tracking-wide`}
```
Comment (vinutv) [inline src/Components/Common/Sidebar/SidebarItem.tsx:65]: https://github.com/ohcnetwork/care_fe/pull/8789#discussion_r1801269426 Fixed here. @sainak

**Comment by nihal467** [comment]
*Bug*

image

- [ ] when user names are long, the text is overflowing and collapse icon is clipping-n in the sidebar

**Comment by vinutv** [comment]
> _Bug_
> 
> image
> 
> * [ ]  when user names are long, the text is overflowing and collapse icon is clipping-n in the sidebar

@nihal467  Implemented truncation for the username. [82966c3](https://github.com/ohcnetwork/care_fe/pull/8789/commits/82966c3f7a6420723cf4e48ba18895a5e1fda29a)

File: src/Components/Common/Sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -219,17 +224,26 @@ interface ToggleShrinkProps {
...
+          className={`flex h-6 w-6 cursor-pointer items-center justify-center rounded focus:outline-none focus:ring-2 focus:ring-indigo-500 ${shrinked ? "bg-gray-200" : "bg-gray-100"} text-gray-600 hover:bg-primary-200 hover:text-primary-800 ${
+            shrinked ? "mx-auto" : "mr-4"
+          } transition-all duration-200 ease-in-out`}
+          onClick={toggle}
+        >
+          <CareIcon
+            icon={shrinked ? "l-arrow-bar-right" : "l-layout-sidebar-alt"}
+            className={`text-lg ${
+              shrinked ? "" : ""
+            } transition-all delay-150 duration-300 ease-out`}
```
Comment (rithviknishad) [inline src/Components/Common/Sidebar/Sidebar.tsx]: @vinutv  do we need this check? it's not applying any classes in both situations right. or was something else intended here?

File: src/Components/Common/Sidebar/SidebarUserCard.tsx [hunk 4]
```diff
@@ -16,47 +23,59 @@ const SidebarUserCard: React.FC<SidebarUserCardProps> = ({ shrinked }) => {
...
+      <DropdownMenu>
+        <DropdownMenuTrigger asChild>
+          <Button
+            variant="outline"
+            size="lg"
+            className={`tooltip relative w-full cursor-pointer items-center justify-between rounded-lg bg-gray-200 p-2 font-normal text-gray-900 transition hover:bg-gray-200 focus:outline-none focus:ring focus:ring-violet-300 ${shrinked ? "flex h-full flex-col-reverse" : "flex flex-row"}`}
+          >
+            <div
+              id="user-profile-name"
+              className={`flex flex-1 items-center justify-start transition ${shrinked ? "" : ""}`}
```
Comment (rithviknishad) [inline src/Components/Common/Sidebar/SidebarUserCard.tsx]: same goes here.

File: src/Components/Common/Sidebar/Sidebar.tsx [hunk 5]
```diff
@@ -132,15 +139,13 @@ const StatelessSidebar = ({
...
             ref={indicatorRef}
-            // className="absolute left-2 w-1 hidden md:block bg-primary-400 rounded z-10 transition-all"
             className={classNames(
-              "absolute left-2 z-10 block w-1 rounded bg-primary-400 transition-all",
-              activeLink ? "opacity-0 md:opacity-100" : "opacity-0",
+              "absolute right-2 z-10 block h-6 w-1 rounded-l bg-primary-500 transition-all",
+              activeLink ? "opacity-100" : "opacity-0",
             )}
           />
           {NavItems.map((i) => {
```
Comment (rithviknishad) [inline src/Components/Common/Sidebar/Sidebar.tsx:160]: `NavItems` texts to be added to i18n 

cc: @nihal467

File: src/Components/Common/Sidebar/Sidebar.tsx (hunk 3 above)
Comment (vinutv) [reply src/Components/Common/Sidebar/Sidebar.tsx]: This check is unnecessary and was missed during cleanup. It has now been removed. [82091c2](https://github.com/ohcnetwork/care_fe/pull/8789/commits/82091c2e3f9d03dd587cf4d187c629d6ccc37161)

File: src/Components/Common/Sidebar/SidebarUserCard.tsx (hunk 4 above)
Comment (vinutv) [reply src/Components/Common/Sidebar/SidebarUserCard.tsx]: [82091c2](https://github.com/ohcnetwork/care_fe/pull/8789/commits/82091c2e3f9d03dd587cf4d187c629d6ccc37161)

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@vinutv @rithviknishad 

image

- [ ] Currently, when the user clicks to open the profile dropdown and then attempts to access the "Asset" tab, the dropdown closes on the initial click, and the "Asset" tab only opens after a subsequent click. The expected behavior is that upon clicking the "Asset" tab, the profile dropdown should automatically close, and the "Asset" tab should open in a single action.
- [x] Clear the merge conflict @rithviknishad

**Comment by vinutv** [comment]
> * [ ]  Currently, when the user clicks to open the profile dropdown and then attempts to access the "Asset" tab, the dropdown closes on the initial click, and the "Asset" tab only opens after a subsequent click. The expected behavior is that upon clicking the "Asset" tab, the profile dropdown should automatically close, and the "Asset" tab should open in a single action.

@nihal467 
Yes, The ideal interaction behaviour is for the new link to open even when the dropdown is still active. 
Could we address this as a separate issue?

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**